### ISSUE TITLE
Writable doesn't only just strip readonly from the keys

### ISSUE THREAD

**Issue by bertearazvan** [sindresorhus/type-fest#717]
Hi! I encounter a weird bug, and I am not sure if I am using it wrong or it's broken somehow.

I am encountering an issue where the `Writable` doesn't only strip the `readonly` but it also changes the object's type structure. Check out the following images for further details. I expect that Writable simply removes `readonly` from the first level of keys in the object.

![Writable1](https://github.com/sindresorhus/type-fest/assets/30385700/2c6f5aba-4828-4abb-b0b1-80d26d9b955c)

![Writable2](https://github.com/sindresorhus/type-fest/assets/30385700/0f86a73e-4720-456d-8ea0-e7c73619e617)

I like the [@utility-types approach to it](https://github.com/piotrwitek/utility-types/blob/411e83ecf70e428b529fc2a09a49519e8f36c8fa/src/mapped-types.ts#L649), it seems simpler with a very similar result. Would you be open to have the same implementation here?

Thanks!

**Comment by sindresorhus** [comment]
Are you on the latest `type-fest` version?

**Comment by sindresorhus** [comment]
> it seems simpler with a very similar result. Would you be open to have the same implementation here?

It's simpler because it doesn't accept an optional type parameter for what keys it should apply to.

I think we could add an overload when that parameter is not passed and use the simple handling.

**Comment by bertearazvan** [comment]
> Are you on the latest `type-fest` version?

Just double checked and yes I am. I have `4.5.0`

**Comment by bertearazvan** [comment]
> I think we could add an overload when that parameter is not passed and use the simple handling.

Yeah, it could be an idea 🚀. I tried it by passing each key as an optional type parameter and it worked as expected.

**Comment by Emiyaaaaa** [comment]
> I think we could add an overload when that parameter is not passed and use the simple handling.

I think the core reason is we wrap the result with `Simplify`
